OG/OG/OG.g4

```diff
@@ -1,83 +1,216 @@
+                | child=factor                                     #singleTermChild //Child er et vidtdækkende begreb
+                ;      
+factor          : lhs=atom pow='^' rhs=factor                      
+                | child=atom                                          
+                |'(' mathExpr=mathExpression ')'
+                ;
+
+atom            : funcCall=functionCall
+                | value=Number 
+                | xyValue=CoordinateXYValue 
+                | id=ID
+                ;
+               
+
+boolExpression  : id=ID                                                    #boolExprID
+                | value=BooleanValue                                          #boolExprTrueFalse
+                | funcCall=functionCall                                          #boolExprFuncCall
+                | lhs=mathExpression BoolOperator rhs=mathExpression    #boolExprMathComp
+                | lhs=boolExpression LogicOperator rhs=boolExpression   #boolExprBoolComp
+                | '!'boolExpr=boolExpression                                     #boolExprNotPrefix   
+                ;
 
-boolExpression  : ID |  BooleanValue |  functionCall 
-                | mathExpression BoolOperator mathExpression 
-                | boolExpression LogicOperator boolExpression 
-                | '!'boolExpression;  //Overvej function call
 
-//a == true er ikke gyldigt!
 
 
 //Commands:
 //Movement commands:
-command         : iterationCommand | movementCommand | drawCommand;
-
-movementCommand : (lineCommand | curveCommand)';';
-lineCommand     : 'line' '.' 'from' '(' (numberTuple | ID | pointReference)')' toCommand+;
-curveCommand    : 'curve' '.' 'withAngle' '(' mathExpression ')' '.' 'from' '('(numberTuple | ID)')' toCommand;
-toCommand       : '.to''(' (numberTuple | ID) ')';
-drawCommand     : ID';' | ID '.'fromCommand;
-fromCommand     :  'from' ( '('ID')' |  '(' numberTuple ')' | '(' StartPointReference ')' | '(' EndPointReference ')'| functionCall );
-
+command         : iterCmd=iterationCommand
+                | movementCmd=movementCommand
+                | drawCmd=drawCommand
+                ;
+
+movementCommand : lineCmd=lineCommand ';' 
+                | curveCmd=curveCommand';'
+                ;
+                
+lineCommand     : 'line' fromCmd=fromCommand  toCmds=toCommands;
+
+toCommands: toCmd=toCommand chainedToCmds=toCommands          #chainedToCommand
+          | toCmd=toCommand                                   #singleToCommand
+          ;
+
+   
+curveCommand    : 'curve''.''withAngle' '('angle=mathExpression ')'  fromCmd=fromCommand toCmd=toCommand;
+                
+toCommand       : '.to''(' id=ID ')'                   #toWithId
+                | '.to''(' tuple=numberTuple ')'          #toWithNumberTuple
+                | '.to''(' toPoint=StartPointReference ')'  #toWithStartPointRef
+                | '.to''(' toPoint=EndPointReference ')'    #toWithEndPointRef
```

Comment:
".to" bør opdeles i "." og "to"

Vores problem er dog, at vi ikke kan enforce, at der ikke må være mellemrum mellem ".to" og "("
På samme måde vil denne ændring medfører, at ".  to   (1,    2 )" er syntaktisk valid. Tror dog ikke det er et stort problem.